### PR TITLE
test: Code Coverage: Add Edit Expense #11

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at 2240824</samp>

This pull request refactors and improves the add edit expense page and its test cases by using new models, methods, and mock data. It also tests various scenarios of expense details with different data and settings. It affects the files `add-edit-expense.page.ts`, `add-edit-expense-*.spec.ts`, `parsed-receipt.data.ts`, and several mock data files in the `core` module.

### <samp>🤖 Generated by Copilot at 2240824</samp>

> _Mock data expands_
> _Testing expense details well_
> _`add-edit-expense`_

### <samp>🤖 Generated by Copilot at 2240824</samp>

*  Add and modify mock data for testing different scenarios in expense details ([link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-95e0540d2a8358265b82fcd223654f17f9166e42c45dc54f4ed0830129f6e34bL55-R69), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-56d972f354883e74d1aa98489079c2d3d173a8d3417fb7d5760cff7a26e7f992R5601-R5605), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8914bfb32bd70efe52e673bc9938855b2c84a494d9464f82c50fe2c8bfbdca45R976-R988), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-1cac010b51c1a7c2fbdf9bf224a7e267779365abe08b0f16df1ef46e41483951R1251-R1287), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-c7abedba9661487444ab115563a479af552ae5d0c5aabfd882e1226a85c6cb94R13), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-c7abedba9661487444ab115563a479af552ae5d0c5aabfd882e1226a85c6cb94R65-R112), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-7d4ac44e75616e1eaa987ab45e21e31f6ff1d9f685abf39023b59a651210204cR437-R459), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-db737e50777751787c68eda6c5be391ed672eb00607a92de6aaf9c945a3010d6L513-R542), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-39e1651d2d3c178069178511c828a9037a699b95b068f31bd927ae2fd23e7d7eR256-R274))
*  Remove unused imports and add missing imports in add edit expense spec files ([link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L2-R6), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L28-R30), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L64-R68), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-c294271c8dad15915f84fbf86ca0233e9317d1c66b356eaef309058076e7fa15L20-R24), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-c294271c8dad15915f84fbf86ca0233e9317d1c66b356eaef309058076e7fa15R33), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-c294271c8dad15915f84fbf86ca0233e9317d1c66b356eaef309058076e7fa15R78-R79), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647R20), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647R29), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L34-R36), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647R44-R45), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L51-R63), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57R766-R812), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-ba304f41b9ceb84ec3edd25c39133e53d238dc01ad0cb7b04fbd999aa2300639L13-R13), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-ba304f41b9ceb84ec3edd25c39133e53d238dc01ad0cb7b04fbd999aa2300639L22-R28), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-ba304f41b9ceb84ec3edd25c39133e53d238dc01ad0cb7b04fbd999aa2300639L41-R53), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-ba304f41b9ceb84ec3edd25c39133e53d238dc01ad0cb7b04fbd999aa2300639L50-R59), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-a763fec878d12e9f0fbac2b5382b030cd01da44167caaf1cff679ba11ffcdfbeL3-R15), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-a763fec878d12e9f0fbac2b5382b030cd01da44167caaf1cff679ba11ffcdfbeR36), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-a763fec878d12e9f0fbac2b5382b030cd01da44167caaf1cff679ba11ffcdfbeL50-R61))
*  Add new test cases for marking an expense as personal if the CCC group id is null ([link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6R880-R910))
*  Add new test cases for getting insta fyle image data with null date, null parsed response and null invoice date ([link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-c294271c8dad15915f84fbf86ca0233e9317d1c66b356eaef309058076e7fa15R400-R430))
*  Add new test cases for generating expense object from form with different values and fields ([link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-c294271c8dad15915f84fbf86ca0233e9317d1c66b356eaef309058076e7fa15R470-R481), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L409-R464), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L427-R505), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L435-R538), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L449-R579), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647R586-R610), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647R616), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L470-R651), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647R657-R681), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L490-R721))
*  Add new test cases for saving an expense in add mode ([link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-c294271c8dad15915f84fbf86ca0233e9317d1c66b356eaef309058076e7fa15R610-R637))
*  Add new test cases for getting autofill category with null recent categories and expense category ([link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L381-R400))
*  Add new test cases for setting up form without currency and amount, without amount and same currency as home currency, and for an expense with different currencies and DRAFT state ([link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647R1140-R1173))
*  Add new test cases for getting new expense observable without complete data url ([link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647R1322-R1345))
*  Add new spies and expectations for the methods that are used to generate expense object from form, check advance account and balance, get CCC settings and get policy expense ([link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-c294271c8dad15915f84fbf86ca0233e9317d1c66b356eaef309058076e7fa15R279), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-c294271c8dad15915f84fbf86ca0233e9317d1c66b356eaef309058076e7fa15R288), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-c294271c8dad15915f84fbf86ca0233e9317d1c66b356eaef309058076e7fa15R306), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-c294271c8dad15915f84fbf86ca0233e9317d1c66b356eaef309058076e7fa15R316), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L409-R464), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L427-R505), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L435-R538), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L449-R579), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647R586-R610), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647R616), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L470-R651), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647R657-R681), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L490-R721), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57R976), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57R1013), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57R1024), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57R1077), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-ba304f41b9ceb84ec3edd25c39133e53d238dc01ad0cb7b04fbd999aa2300639R910-R1089), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-ba304f41b9ceb84ec3edd25c39133e53d238dc01ad0cb7b04fbd999aa2300639R1199-R1201), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-ba304f41b9ceb84ec3edd25c39133e53d238dc01ad0cb7b04fbd999aa2300639R1392-R1394), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-ba304f41b9ceb84ec3edd25c39133e53d238dc01ad0cb7b04fbd999aa2300639L1256-R1449), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-ba304f41b9ceb84ec3edd25c39133e53d238dc01ad0cb7b04fbd999aa2300639R1559))
*  Modify the mock data and test cases for getting autofill category and setting up form for a CCC expense to test different scenarios ([link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647L373-R378), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-ba304f41b9ceb84ec3edd25c39133e53d238dc01ad0cb7b04fbd999aa2300639L1165-R1357), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-ba304f41b9ceb84ec3edd25c39133e53d238dc01ad0cb7b04fbd999aa2300639L1179-R1371))
*  Add new imports for UnflattenedTransaction and ExtendedAccount models in the add edit expense page and spec file ([link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57R766-R812), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bR109-R110))
*  Add new methods for checking advance account and balance, getting CCC settings and getting different fields from the form group in the add edit expense page (  [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bR2524-R2546))
*  Refactor the filter logic for accounts in the getPaymentModes method to use the new methods for checking advance account and balance and getting CCC settings in the add edit expense page (  [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL2466-R2470))
*  Refactor the subscription logic for org settings and etxn in the add edit expense page to use the new methods for getting CCC settings and different fields from the form group ([link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL2613-R2637), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL2806-R2833), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bR2873-R2974))
*  Refactor the generateEtxnFromFg method in the add edit expense page to use the new methods for getting different fields from the form group ([link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL2900-R3024), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL2910-R3048), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL2937-R3067))
*  Refactor the `criticalPolicyViolated` variable to use a helper method `getIsPolicyExpense` that checks if an expense violates any critical policy in the add edit expense page ([link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL3344-R3465), [link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL3372-R3493))
*  Fix a typo in the property name of the selectedCCCTransaction object in the add edit expense page ([link](https://github.com/fylein/fyle-mobile-app/pull/2213/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL2499-R2500))

## Clickup
https://app.clickup.com/t/85ztk93kr

## Code Coverage
`98.44% Statements 1459/1482`
`95.2% Branches 1389/1459`
`97.87% Functions 460/470`
`98.67% Lines 1415/1434`

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

**Review by Dimple16 — CHANGES_REQUESTED** [review]
I’ll review once the unit test issues are resolved, please fix and reassign

File: src/app/core/mock-data/expense.data.ts [hunk 1]
```diff
@@ -5598,3 +5598,8 @@ export const expectedFormattedTransaction: Partial<Expense>[] = [
     tx_locations: [],
   },
 ];
+
+export const policyExpense2 = {
```
Comment (Dimple16) [inline src/app/core/mock-data/expense.data.ts:5602]: type?

File: src/app/core/mock-data/corporate-card-expense-unflattened.data.ts [hunk 2]
```diff
@@ -52,5 +52,18 @@ export const expectedECccResponse: CCCExpUnflattened[] = [
...
 export const eCCCData1: CCCExpUnflattened = {
   ...expectedECccResponse[0],
+  ccce: {
+    ...expectedECccResponse[0].ccce,
+    corporate_credit_card_account_number: '123456789',
+  },
   flow: 'newCCCFlow',
 };
+
+export const eCCCData2 = {
```
Comment (Dimple16) [inline src/app/core/mock-data/corporate-card-expense-unflattened.data.ts:62]: type?

File: src/app/core/mock-data/org-settings.data.ts [hunk 3]
```diff
@@ -1248,3 +1248,40 @@ export const taxSettingsData2 = {
   ...taxSettingsData,
   enabled: false,
 };
+
```
Comment (Dimple16) [inline src/app/core/mock-data/org-settings.data.ts:1251]: please add types to all the mock data objects

File: src/app/core/mock-data/parsed-receipt.data.ts [hunk 4]
```diff
@@ -61,3 +62,51 @@ export const instaFyleData2 = {
     vendor_name: 'vendor',
   },
 };
+
```
Comment (Dimple16) [inline src/app/core/mock-data/parsed-receipt.data.ts:65]: types are missing in this file too

File: src/app/fyle/add-edit-expense/add-edit-expense-2.spec.ts [hunk 5]
```diff
@@ -426,6 +467,18 @@ export function TestCases2(getTestBed) {
           done();
         });
       });
+
+      it('shuld return txn without category', (done) => {
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/add-edit-expense-2.spec.ts:471]: `shuld return txn without category` - typo
when does this happen though? add it to the description

File: src/app/core/mock-data/unflattened-txn.data.ts [hunk 6]
```diff
@@ -1681,6 +1681,17 @@ export const unflattenedExpenseWithCCCGroupId = {
   },
 };
 
+export const unflattenedExpenseWithCCCGroupId2 = {
```
Comment (Dimple16) [inline src/app/core/mock-data/unflattened-txn.data.ts:1684]: types are missing for most of the mock data in this file, please make sure you add them/create if not present already

File: src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts [hunk 7]
```diff
@@ -446,37 +549,136 @@ export function TestCases3(getTestBed) {
...
+            expect(component.getDistance).toHaveBeenCalledTimes(1);
+            expect(component.getDistanceUnit).toHaveBeenCalledTimes(1);
+            expect(component.getBreakfastProvided).toHaveBeenCalledTimes(1);
+            expect(component.getDuplicateReason).toHaveBeenCalledTimes(1);
+            expect(component.getAmount).toHaveBeenCalledTimes(1);
             done();
           });
       });
 
       it('should generate expense from form without cost center and location data and not a policy expense in edit mode', (done) => {
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts:656]: `should generate expense from form without cost center and location data and not a policy expense in edit mode`
do you mean without policy violation? please rephrase this description, didn't understand it

File: src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts [hunk 8]
```diff
@@ -910,6 +1137,40 @@ export function TestCases3(getTestBed) {
           done();
         });
       });
+
+      it('should get category if initial fetch is false', (done) => {
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts:1141]: `should get category if initial fetch is false` initial fetch meaning?

File: src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts [hunk 9]
```diff
@@ -910,6 +1137,40 @@ export function TestCases3(getTestBed) {
...
+
+        component.getCategoryOnEdit(orgCategoryData[0]).subscribe((res) => {
+          expect(res).toBeUndefined();
+          expect(orgUserSettingsService.get).toHaveBeenCalledTimes(1);
+          expect(orgSettingsService.get).toHaveBeenCalledTimes(1);
+          done();
+        });
+      });
+
+      it('should return null if not category could be found', (done) => {
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts:1158]: `should return null if no category is found`?

File: src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts [hunk 10]
```diff
@@ -1058,6 +1319,30 @@ export function TestCases3(getTestBed) {
           done();
         });
       });
+
+      it('should get new expense observable without complete data url', (done) => {
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts:1323]: `should get new expense observable without insta fyle image data URL?`

File: src/app/fyle/add-edit-expense/add-edit-expense-4.spec.ts [hunk 11]
```diff
@@ -763,6 +763,53 @@ export function TestCases4(getTestBed) {
         expect(router.navigate).toHaveBeenCalledOnceWith(['/', 'enterprise', 'personal_cards']);
         expect(trackingService.newExpenseCreatedFromPersonalCard).toHaveBeenCalledOnceWith();
       });
+
+      it('should match an expense while offline', () => {
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/add-edit-expense-4.spec.ts:767]: please mention that the expense is also being created here

File: src/app/fyle/add-edit-expense/add-edit-expense-4.spec.ts [hunk 12]
```diff
@@ -763,6 +763,53 @@ export function TestCases4(getTestBed) {
...
+        expect(component.generateEtxnFromFg).toHaveBeenCalledTimes(2);
+        expect(transactionService.upsert).toHaveBeenCalledOnceWith(unflattenedTransactionDataPersonalCard);
+        expect(personalCardsService.matchExpense).toHaveBeenCalledOnceWith(
+          unflattenedTransactionDataPersonalCard.split_group_id,
+          apiPersonalCardTxnsRes.data[0].btxn_id
+        );
+        expect(component.uploadAttachments).toHaveBeenCalledOnceWith(
+          unflattenedTransactionDataPersonalCard.split_group_id
+        );
+        expect(component.showSnackBarToast).toHaveBeenCalledOnceWith(
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/add-edit-expense-4.spec.ts:805]: are we writing a test case for failure case too? or is it already handled?

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 13]
```diff
@@ -2533,6 +2533,29 @@ export class AddEditExpensePage implements OnInit {
       });
   }
 
+  getIsPolicyExpense(etxn: Expense) {
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2536]: what are the return types for the below methods?

**Review by Dimple16 — CHANGES_REQUESTED** [review]
added comments

File: src/app/core/mock-data/parsed-receipt.data.ts (hunk 4 above)
Comment (jayfyle) [reply src/app/core/mock-data/parsed-receipt.data.ts:65]: Added in linting PR

File: src/app/core/mock-data/unflattened-txn.data.ts (hunk 6 above)
Comment (jayfyle) [reply src/app/core/mock-data/unflattened-txn.data.ts:1684]: Adding types for as many as possible, will add the rest in linting PR, where types for functions are clearer

File: src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts (hunk 8 above)
Comment (jayfyle) [reply src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts:1141]: initial fetch is a class property, that is false by default and is set to true when custom fields are setup.

File: src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts (hunk 9 above)
Comment (jayfyle) [reply src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts:1158]: Rephrased the spec description, it returns null if expense has no category and auto-fill category is also null

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts (hunk 13 above)
Comment (jayfyle) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:2536]: Added in the linting PR

File: src/app/fyle/add-edit-expense/add-edit-expense-4.spec.ts (hunk 12 above)
Comment (jayfyle) [reply src/app/fyle/add-edit-expense/add-edit-expense-4.spec.ts:805]: I have added cases for violations but yes error case is missing due to some issue causing tests to fail.

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 14]
```diff
@@ -936,25 +936,28 @@ export class AddEditExpensePage implements OnInit {
     );
   }
 
+  checkAdvanceAccountAndBalance(account: ExtendedAccount) {
```
Comment (Julias0) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:939]: add return type

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 15]
```diff
@@ -936,25 +936,28 @@ export class AddEditExpensePage implements OnInit {
...
-            )
+            map((accounts) => accounts.filter((account) => this.checkAdvanceAccountAndBalance(account)).length > 0)
           );
         }
         return of(false);
       })
     );
   }
 
+  getCCCSettings(orgSettings) {
```
Comment (Julias0) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:957]: add return type

**Review by Julias0 — APPROVED** [review]
